src/modules/awake/Awake/Core/APIHelper.cs

```diff
@@ -99,9 +102,16 @@ public static void SetIndefiniteKeepAwake(Action<bool> callback, Action failureC
             _tokenSource = new CancellationTokenSource();
             _threadToken = _tokenSource.Token;
 
-            _runnerThread = Task.Run(() => RunIndefiniteLoop(keepDisplayOn), _threadToken)
-                .ContinueWith((result) => callback(result.Result), TaskContinuationOptions.OnlyOnRanToCompletion)
-                .ContinueWith((result) => failureCallback, TaskContinuationOptions.NotOnRanToCompletion);
+            try
+            {
+                _runnerThread = Task.Run(() => RunIndefiniteLoop(keepDisplayOn), _threadToken)
+                    .ContinueWith((result) => callback(result.Result), TaskContinuationOptions.OnlyOnRanToCompletion)
+                    .ContinueWith((result) => failureCallback, TaskContinuationOptions.NotOnRanToCompletion);
+            }
+            catch (Exception ex)
```

Comment:
Is the idea here to catch errors from starting the task or from the actual task (like the `RunIndefiniteLoop` call?)
I think this will only catch errors from starting the task.